### PR TITLE
an immature version just to make sure I'm on the right track

### PR DESCRIPTION
**Pull request by Zee314159**
By the way, what kind of message would users be interested, and what kind of message would be probably just garbage?

### CODE REVIEW

**Comment by patrickvacek** [comment]
Thanks! The idea is basically right: we just need simple confirmation of what the tool is actually doing. But I would suggest doing it at a higher level, possibly in `src/aktualizr_repo/main.cc` directly. There's a big switch that starts with `if (command == "generate")` and each command should just print a simple line of output. So for "generate", at the end it should print something like "Uptane metadata repos generated at <path>".

**Comment by Zee314159** [comment]
> Uptane metadata repos generated at some-path

Thanks, got it! But somehow there appears to be some delay of your comment... or maybe I should just refresh browser several times more.

**Comment by patrickvacek** [comment]
> somehow there appears to be some delay of your comment... or maybe I should just refresh browser several times more.

What was the delay? I'm not sure I understand.

**Comment by Zee314159** [comment]
> somehow there appears to be some delay of your comment... or maybe I should just refresh browser several times more.
> 
> What was the delay? I'm not sure I understand.

I checked this PR in the morning, like 6 or 7 hours ago, and I didn't see your comment. But when I checked again in the afternoon, maybe 1 hour ago, I found your comment was made 17 hours ago. Not sure what happened, maybe it's a browser issue. Anyway I am adding messages in _main.cc_ now, probably will finish soon. I'll push the changes later.

**Comment by patrickvacek** [comment]
> I checked this PR in the morning, like 6 or 7 hours ago, and I didn't see your comment. But when I checked again in the afternoon, maybe 1 hour ago, I found your comment was made 17 hours ago.

Ah, that is annoying. I often have similar issues with github, and a hard refresh is often necessary. I usually avoid this entirely by using github's watch feature so that I get automatic emails about comments and such. It probably makes sense for you to do the same.

> Anyway I am adding messages in _main.cc_ now, probably will finish soon. I'll push the changes later.

Sounds good, thanks!

**Comment by Zee314159** [comment]
> I often have similar issues with github, and a hard refresh is often necessary. I usually avoid this entirely by using github's watch feature so that I get automatic emails about comments and such. It probably makes sense for you to do the same.

So, it's a github issue. Thank you for your tips!

File: src/aktualizr_repo/main.cc [hunk 1]
```diff
@@ -148,18 +154,23 @@ int main(int argc, char **argv) {
         repo.addDelegation(Uptane::Role(vm["dname"].as<std::string>(), true),
                            Uptane::Role(dparent, dparent != "targets"), vm["dpattern"].as<std::string>(),
                            vm["dterm"].as<bool>(), key_type);
+        std::cout << "added a delegated role" << std::endl;
```
[the remark this answers is not in the document]
Comment (Zee314159) [reply src/aktualizr_repo/main.cc]: Cool, thanks! I'll fix it right away.

**Comment by Zee314159** [comment]
> I recall when we talked earlier you mentioned a question or concern with the delegations. It looks like you're on the right track with them. Was there still something unclear with them?

I tested the delegation command line, and  it gives the message I expect. No more questions for the moment.

File: src/aktualizr_repo/main.cc [hunk 2]
```diff
@@ -130,36 +138,47 @@ int main(int argc, char **argv) {
             custom["targetFormat"] = vm["targetformat"].as<std::string>();
           }
           repo.addCustomImage(targetname.string(), *hash, vm["targetlength"].as<uint64_t>(), delegation, custom);
+          std::cout << "Added a target " << targetname.string() << " to the custom image" << std::endl;
```
Comment (patrickvacek) [inline src/aktualizr_repo/main.cc]: Minor issue, but the custom image IS the target. Might be better to phrase this as "Added a custom image target xyz".

**Pull request merged**